### PR TITLE
feat(api): standardized titles of error messages

### LINKED ISSUE #4017: Error handling repeats title and details
For example a 404: 

```JSON
{"error":
[{"status":"404",
"title":"Results could not be found for simulation run 'xxx'.",
"detail":"\"Results could not be found for simulation run 'xxx'.\""}]
```
The title should be 'Not Found'

### PR DESCRIPTION
**Pull request by jonrkarr**
Closes #4017 

Standardized titles using the `http-status-codes` package.

@bilalshaikh42 the function below seems to be unused. Please confirm this can be deleted.

### CODE REVIEW

File: libs/shared/exceptions/exceptions/src/lib/exception.ts [hunk 1]
```diff
@@ -51,35 +63,22 @@ export class BiosimulationsException extends Error {
...
   getStatus(): number {
     return this.status;
   }
+
   getError(): ErrorObject {
     return this.errorObject;
   }
-
-  // TODO write test
-  static fromHTTP(exception: HttpException): BiosimulationsException {
```
Comment (jonrkarr) [inline libs/shared/exceptions/exceptions/src/lib/exception.ts:67]: @bilalshaikh42 this isn't used anywhere. Please confirm this can be deleted.

Comment (bilalshaikh42) [reply libs/shared/exceptions/exceptions/src/lib/exception.ts:67]: This was meant to be added to the error handlers for the HTTP exception filter to provide a consistent error object. If it is not used that would be the cause of #4017

Comment (bilalshaikh42) [reply libs/shared/exceptions/exceptions/src/lib/exception.ts:67]: With the rest of the changes, this seems to be redundant now.

**Review by bilalshaikh42 — APPROVED** [review]

File: libs/shared/exceptions/exceptions/src/lib/exception.ts (hunk 1 above)
Comment (jonrkarr) [reply libs/shared/exceptions/exceptions/src/lib/exception.ts:67]: That method was never hooked up to anything. The HTTP errors and BioSimulationsExceptions seem to be separate. I decided to standardize the titles of both by using the numeric codes to look up standardized titles.

**Pull request merged**